map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java

```diff
@@ -72,82 +69,148 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
+      // create an ArcGISMap with the topographic basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+
+      // create a map view and set the ArcGISMap to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set the initial viewpoint for the map view
+      mapView.setViewpoint(new Viewpoint(new Point(-11e6, 45e5, SpatialReferences.getWebMercator()), 40000000));
+
       // create a control panel
-      VBox controlsVBox = new VBox(6);
-      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.3)"), CornerRadii.EMPTY,
-          Insets.EMPTY)));
-      controlsVBox.setPadding(new Insets(10.0));
-      controlsVBox.setMaxSize(220, 90);
+      controlsVBox = new VBox();
       controlsVBox.getStyleClass().add("panel-region");
 
-      // create labels to display the view status of each layer
-      Label worldTimeZonesLabel = new Label("World Time Zones: ");
-      Label censusLabel = new Label("Census: ");
-      Label facilitiesLabel = new Label("Facilities: ");
+      // create a label to display the view status and add to the control panel
+      layerViewStatusLabel = new Label("Click button to load feature layer\n ");
+      layerViewStatusLabel.getStyleClass().add("panel-label");
+      controlsVBox.getChildren().add(layerViewStatusLabel);
 
-      worldTimeZonesLabel.getStyleClass().add("panel-label");
-      censusLabel.getStyleClass().add("panel-label");
-      facilitiesLabel.getStyleClass().add("panel-label");
+      // create a listener that fires every time a layer's view status changes
+      mapView.addLayerViewStateChangedListener(statusChangeEvent -> {
 
-      // add labels to the control panel
-      controlsVBox.getChildren().addAll(worldTimeZonesLabel, censusLabel, facilitiesLabel);
-
-      // create three layers to add to the ArcGISMap
-      final ArcGISTiledLayer tiledLayer = new ArcGISTiledLayer(SERVICE_TIME_ZONES);
+        // get the layer whose state has changed
+        Layer layer = statusChangeEvent.getLayer();
+        // only update the status if the layer is the feature layer being tracked
+        if (layer != featureLayer) {
+          return;
+        }
+        // get the layer's view status and display the status
+        EnumSet<LayerViewStatus> layerViewStatus = statusChangeEvent.getLayerViewStatus();
+        displayViewStateText(layerViewStatus);
+
+        // if there is an error or warning, format and display the message as an alert
+        ArcGISRuntimeException error = statusChangeEvent.getError();
+        if (error != null) {
+          Throwable cause = error.getCause();
+          String message = (cause != null) ? cause.toString() : error.toString();
```

Comment:
On this one, there was a bit of discussion when this was initially implemented for the design. There's a different message depending on what the error is - requesting too many features above the max query count or due to the network issues. So the layer view status itself updates to WARNING or ERROR but there is an additional error message that accompanies it.

The 'true' method in the ternary (`cause.toString`) is the one that prints out as a Java Exception when there is a network issue, the false option (should be `error.getMessage` which I'll fix..) prints out nicely for when there's too many feature requests, according to the design discussions. 

So maybe, I could replace first message to be a manual string instead of the UnknownHostException that displays, and leave the other one as the default message "A layer has requested more features than the service maximum feature count". Thoughts?

Reply:
That sounds like a good solution, yeah!